app/integration/runtime/src/main/java/io/syndesis/integration/runtime/IntegrationRouteBuilder.java

```diff
@@ -287,23 +287,18 @@ private void configureFlow(Flow flow, final String flowIndex) throws URISyntaxEx
             }
         } else {
             // not in a pipeline so set the step id header to be sure it is there
-            definition = parent.setHeader(IntegrationLoggingConstants.STEP_ID, constant(stepId));
+            parent.setHeader(IntegrationLoggingConstants.STEP_ID, constant(stepId));
         }
 
-        definition = parent.process(OutMessageCaptureProcessor.INSTANCE)
+        return parent.process(OutMessageCaptureProcessor.INSTANCE)
                         .id(String.format("capture-out:%s", stepId));
-        return definition;
```

Comment:
definition variable was not really used. We can avoid assigning and preventing weird behaviours in the future.